main.go

```diff
@@ -3,15 +3,23 @@ package main
 import (
 	"os"
 
-	"github.com/hashicorp/terraform-ls/commands"
-	"github.com/hashicorp/terraform-ls/version"
 	"github.com/mitchellh/cli"
+
+	"github.com/hashicorp/terraform-ls/commands"
+)
+
+var (
+	// These vars will be set by goreleaser.
+	version string = `dev`
+	commit  string = ``
+	date    string = ``
+	builtBy string = ``
```

Comment:
As discussed offline I'm ok with these variables being in `main` package - as we can pass them down to wherever they're needed and it is better than global state in `version` package.

However (1) I think it'd be cleaner to still keep them in a separate file and (2) it would be better to retain some of the logic from the `version` package:

 - variable names `version`/`prerelease` - which can be un-exported, but I think we should retain the meaning and avoid setting `version` to `dev`. I'd say the meaningful default is `version = "0.0.0"` `prerelease = "dev"`
 - semver check via `init()` - which won't be `panic`ing on `go get/build/install` anymore if `version` defaults to `0.0.0`, but still gives us safety net against building a version which isn't semver-compatible

Reply:
A separate file is a good call.

Why have separate fields? It gets complicated for snapshots as its already going to send a prerelease string to the version (ie. `1.2.3-SNAPSHOT-hash`) or if your goreleaser tag is prerelease, it will mess it up wouldn't it? As the `version` var and the `prerelease` var will kinda be redundant. Could just do `version string = "0.0.0-dev"` couldn't you?

Reply:
As for GoReleaser - it may do by default, but it won't with custom ldflags, like we have now (without this patch), except that these flags would be less verbose with variables being in `main`.

As for why - I don't have any better reason other than keeping it aligned with other HashiCorp Go projects ([terraform](https://github.com/hashicorp/terraform/blob/master/version/version.go#L19), [packer](https://github.com/hashicorp/packer/blob/master/version/version.go#L17), [consul](https://github.com/hashicorp/consul/blob/master/version/version.go#L23), [vault](https://github.com/hashicorp/vault/blob/master/sdk/version/version.go#L12), [nomad](https://github.com/hashicorp/nomad/blob/master/version/version.go#L19)). 

That said I appreciate you challenging this convention and I have not asked teams _why_ they do it and _if_ they still feel this is right in today's context, as the separate package thing probably isn't in retrospect and this may be just a relic of some inflexible release tooling we may still have in place.

So I'll ask around next week - and if there are no strong opinions/reasons then I'm happy to go ahead with the "GoReleaser style" you're proposing! 😄